clang/lib/AST/Expr.cpp

```diff
@@ -4247,6 +4247,13 @@ FieldDecl *Expr::getSourceBitField() {
     if (UnOp->isPrefix() && UnOp->isIncrementDecrementOp())
       return UnOp->getSubExpr()->getSourceBitField();
 
+  if (const ConditionalOperator *Cond = dyn_cast<ConditionalOperator>(E)) {
+    if (FieldDecl *FD = Cond->getTrueExpr()->getSourceBitField())
+      return FD;
+    if (FieldDecl *FD = Cond->getFalseExpr()->getSourceBitField())
+      return FD;
+  }
```

Comment:
Wouldn't this mean the order of the second and third operands would matter? Also, it looks like it would make cond?bit-field:non-bit-field promote and check for narrowing conversions based off of the width of the bit-field. Therefore promotion could change the value if the non-bit-field had a value not representable in int but representable in unsigned, and narrowing could happen in contexts where narrowing conversions are disabled.